control/control.go

```diff
@@ -487,26 +494,66 @@ func (c *Controller) Solve(ctx context.Context, req *controlapi.SolveRequest) (*
 }
 
 func (c *Controller) Status(req *controlapi.StatusRequest, stream controlapi.Control_StatusServer) error {
+	var buildID string
+	ctx := stream.Context()
+	peer, ok := peer.FromContext(ctx)
+	if ok {
+		tlsInfo, ok := peer.AuthInfo.(credentials.TLSInfo)
+		if ok && tlsInfo.SPIFFEID != nil {
+			buildID = tlsInfo.SPIFFEID.String()
+		}
+	}
+
 	if err := sendTimestampHeader(stream); err != nil {
 		return err
 	}
 	ch := make(chan *client.SolveStatus, 8)
 
-	eg, ctx := errgroup.WithContext(stream.Context())
+	eg, ctx := errgroup.WithContext(ctx)
 	eg.Go(func() error {
 		return c.solver.Status(ctx, req.Ref, ch)
 	})
 
 	eg.Go(func() error {
 		for {
+			var sender *connect.ClientStreamForClient[cloudv3.ReportStatusRequest, cloudv3.ReportStatusResponse]
+			token := os.Getenv("DEPOT_BUILDKIT_TOKEN")
+			if buildID != "" && token != "" {
+				depotclient := NewDepotClient()
+				sender = depotclient.ReportStatus(ctx)
+				sender.RequestHeader().Add("Authorization", "Bearer "+token)
+			}
+
 			ss, ok := <-ch
 			if !ok {
+				if sender != nil {
+					_, err := sender.CloseAndReceive()
+					if err != nil {
+						bklog.G(ctx).WithError(err).Errorf("unable to close status sender")
+					}
+				}
+
 				return nil
 			}
+
 			for _, sr := range ss.Marshal() {
 				if err := stream.SendMsg(sr); err != nil {
 					return err
 				}
+				if sender != nil {
+					stableDigests := make(map[string]string, len(sr.Vertexes))
+					for _, v := range sr.Vertexes {
+						stableDigests[v.Digest.String()] = v.StableDigest.String()
+					}
+					req := &cloudv3.ReportStatusRequest{
+						BuildId:       buildID,
```

Comment:
Nice, I think we should probably rename this field to `SpiffeId`, in the API too, since that's what it would be. The actual format of the ID will probably be something like `spiffe://depot.dev/org/ORG/project/PROJ/build/BUILD` but it'll be opaque to BuildKit.

Reply:
Ok! I force pushed it in [1355541](https://github.com/depot/buildkit/pull/41/commits/135554106b199d9450e2a0646062acef13036ef9)